dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java

```diff
@@ -114,36 +114,54 @@ private Optional<PageInfo> getPageInfo(Path path) {
         createdTime = creationTime.toMillis();
       } catch (IOException e) {
         LOG.error("Failed to get file size for " + path, e);
+        deleteUnrecognizedPage(path);
         return Optional.empty();
       }
       return Optional.of(new PageInfo(pageId.get(),
           pageSize, CacheScope.GLOBAL, this, createdTime));
     }
+    deleteUnrecognizedPage(path);
     return Optional.empty();
   }
 
+  /**
+   * Deletes an unrecognized page file due to various reasons.
+   * @param path the file path of a page file
+   */
+  private void deleteUnrecognizedPage(Path path) {
+    try {
+      Files.delete(path);
```

Comment:
add a precondition check that path starts with `getRootPath()` to avoid deleting anything outside of the cache directory.

Reply:
Good point.

Reply:
This is not a public function which will be called from outside of this class.
The path is from the Files.walk() in scanPages(), which is under the `getRootPath()` dir.

But, it's a good practice to add precondition checking.

Reply:
Oh i see. then it's fine to leave out the precondition check.